ng/src/web/pages/filters/dialog.js

```diff
@@ -23,89 +24,110 @@
 
 import React from 'react';
 
-import  _ from 'gmp/locale.js';
+import _ from 'gmp/locale.js';
 
 import Layout from '../../components/layout/layout.js';
 
 import PropTypes from '../../utils/proptypes.js';
 
-import withDialog from '../../components/dialog/withDialog.js';
+import SaveDialog from '../../components/dialog/savedialog.js';
 
 import FormGroup from '../../components/form/formgroup.js';
 import TextField from '../../components/form/textfield.js';
 import Select from '../../components/form/select.js';
 
+const DEFAULTS = {name: _('Unnamed'), term: ''};
+
 const FilterDialog = ({
-    comment,
-    name,
+    filter,
     term,
+    title = _('New Filter'),
     type,
     types,
-    onValueChange,
+    visible,
+    onClose,
+    onSave,
   }) => {
+
   return (
-    <Layout flex="column">
-      <FormGroup title={_('Name')}>
-        <TextField
-          name="name"
-          grow="1"
-          value={name}
-          size="30"
-          onChange={onValueChange}
-          maxLength="80"/>
-      </FormGroup>
-
-      <FormGroup title={_('Comment')}>
-        <TextField
-          name="comment"
-          grow="1"
-          value={comment}
-          size="30"
-          maxLength="400"
-          onChange={onValueChange}/>
-      </FormGroup>
-
-      <FormGroup title={_('Term')}>
-        <TextField
-          name="term"
-          grow="1"
-          value={term}
-          size="30"
-          onChange={onValueChange}/>
-      </FormGroup>
-
-      <FormGroup title={_('Type')}>
-        <Select
-          name="type"
-          onChange={onValueChange}
-          value={type}>
-          {
-            types.map(option => (
-              <option key={option[1]} value={option[1]}>{option[2]}</option>
-            ))
-          }
-        </Select>
-      </FormGroup>
-    </Layout>
+    <SaveDialog
+      visible={visible}
+      title={title}
+      onClose={onClose}
+      onSave={onSave}
+      initialData={{
+        ...DEFAULTS,
+        ...filter,
+        term,
+        type,
+      }}
+    >
+      {({
+        data: state,
+        onValueChange,
+      }) => {
+        const filterOptions = types.map(option => ({
```

Comment:
Would be better to move the filterOptions creation to the top before the first jsx statement. In that case it will be only re-created when the dialog is re-rendered. Currently it is re-created on each input value change which is very often.